src/sirfilesystem.c

```diff
@@ -182,9 +182,9 @@ char* _sir_getappfilename(void) {
 # endif
     struct stat st;
     if (-1 == lstat(PROC_SELF, &st)) {
-        _sir_handleerr(errno);
+        _sir_handleerr(errno); // GCOVR_EXCL_START
```

Comment:
We could write a test on these platforms to break this and trigger this code path rather than cheating